Discrepancies in column types for social-auth-core derived database tables

**Describe the bug**
social-auth-core 4.5.5 has added initial type annotation, which is [breaking our test_galaxy_packages tests](https://github.com/galaxyproject/galaxy/actions/runs/13326332291/job/37220297670) (since these tests don't pin a version of the library).
The failing mypy tests are all about column types for database table inheriting from social-auth-core `*Mixin` classes. Some failures seem legitimate errors in our column type definitions, others may be discrepancies we can ignore with a `type: ignore`, and some may be errors in type upstream type annotation:

```
galaxy/model/__init__.py:10152: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "AssociationMixin" defined
the type as "str")  [assignment]
        server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10153: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "AssociationMixin" defined
the type as "str")  [assignment]
        handle: Mapped[Optional[str]] = mapped_column(VARCHAR(255))
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10154: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "AssociationMixin" defined
the type as "str")  [assignment]
        secret: Mapped[Optional[str]] = mapped_column(VARCHAR(255))
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10155: error: Incompatible types in assignment
(expression has type "Mapped[int | None]", base class "AssociationMixin" defined
the type as "int")  [assignment]
        issued: Mapped[Optional[int]]
        ^
galaxy/model/__init__.py:10156: error: Incompatible types in assignment
(expression has type "Mapped[int | None]", base class "AssociationMixin" defined
the type as "int")  [assignment]
        lifetime: Mapped[Optional[int]]
        ^
galaxy/model/__init__.py:10157: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "AssociationMixin" defined
the type as "str")  [assignment]
        assoc_type: Mapped[Optional[str]] = mapped_column(VARCHAR(64))
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10216: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "CodeMixin" defined the
type as "str")  [assignment]
        email: Mapped[Optional[str]] = mapped_column(VARCHAR(200))
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10217: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "CodeMixin" defined the
type as "str")  [assignment]
        code: Mapped[Optional[str]] = mapped_column(VARCHAR(32))
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10244: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "NonceMixin" defined the
type as "str")  [assignment]
        server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10245: error: Incompatible types in assignment
(expression has type "Mapped[int | None]", base class "NonceMixin" defined the
type as "int")  [assignment]
        timestamp: Mapped[Optional[int]]
        ^
galaxy/model/__init__.py:10246: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "NonceMixin" defined the
type as "str")  [assignment]
        salt: Mapped[Optional[str]] = mapped_column(VARCHAR(40))
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10282: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "PartialMixin" defined the
type as "str")  [assignment]
        token: Mapped[Optional[str]] = mapped_column(VARCHAR(32))
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10283: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "PartialMixin" defined the
type as "dict[Any, Any]")  [assignment]
        data: Mapped[Optional[str]] = mapped_column(TEXT)
                                      ^~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10284: error: Incompatible types in assignment
(expression has type "Mapped[int | None]", base class "PartialMixin" defined the
type as "str")  [assignment]
        next_step: Mapped[Optional[int]]
        ^
galaxy/model/__init__.py:10285: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "PartialMixin" defined the
type as "str")  [assignment]
        backend: Mapped[Optional[str]] = mapped_column(VARCHAR(32))
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10327: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "UserMixin" defined the
type as "None")  [assignment]
        uid: Mapped[Optional[str]] = mapped_column(VARCHAR(255))
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10328: error: Incompatible types in assignment
(expression has type "Mapped[str | None]", base class "UserMixin" defined the
type as "str")  [assignment]
        provider: Mapped[Optional[str]] = mapped_column(VARCHAR(32))
                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
galaxy/model/__init__.py:10332: error: Incompatible types in assignment
(expression has type "Mapped[User | None]", base class "UserMixin" defined the
type as "str")  [assignment]
        user: Mapped[Optional["User"]] = relationship(back_populates="soci...
                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~...
```

For reference, you can look at the reference definition of these tables from the [social-auth-storage-sqlalchemy](https://github.com/python-social-auth/social-storage-sqlalchemy) package: https://github.com/python-social-auth/social-storage-sqlalchemy/blob/master/social_sqlalchemy/storage.py

Initial assessment:
- Class `PSAAssociation(AssociationMixin)`:
   - columns `server_url`, `handle`, `secret`, `issued`, `lifetime`, `assoc_type`: we could drop `Optional`
- Class `PSACode(CodeMixin)`:
   - columns `email`, `code`: we could drop `Optional`
- Class `PSANonce(NonceMixin)`:
   - columns `server_url`, `timestamp`, `salt`: we could drop `Optional`
- Class `PSAPartial(PartialMixin)`:
   - columns: `token`, `backend`: we could drop `Optional`
   - column `data`: I think this should be changed from `Mapped[Optional[str]] = mapped_column(TEXT)` to `Mapped[[Dict[str, Any]] = mapped_column(MutableJSONType)`
   - column `next_step`: we could drop `Optional`. Need to fix upstream type from `str` to `int`
- Class `UserAuthnzToken(UserMixin)`:
   - columns `uid`, `provider`: we could drop `Optional`
   - column `user`: drop the `user` attribute of the `UserMixin` class upstream

@jdavcs @dannon Thoughts?

Apparently, it's not solvable with `nullable=True'
```
galaxy/model/__init__.py:10087: error: Incompatible types in assignment (expression has type "Mapped[str]", base class "AssociationMixin" defined the type as "str")  [assignment]
        server_url: Mapped[str] = mapped_column(VARCHAR(255), nullable=True)
```
I'll look some more, but it may be an upstream problem. In which case I suggest going with the type-ignore, at least for now.

--- a/lib/galaxy/model/__init__.py
+++ b/lib/galaxy/model/__init__.py
@@ -10084,7 +10084,7 @@ class PSAAssociation(Base, AssociationMixin, RepresentById):
     __tablename__ = "psa_association"
 
     id: Mapped[int] = mapped_column(primary_key=True)
-    server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
+    server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type: ignore[assignment, unused-ignore]  # needed for social-auth-core Mixin class attributes. unused-ignore needed for older social-auth-core on Python 3.8
     handle: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
     secret: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
     issued: Mapped[Optional[int]]  # type:ignore[assignment, unused-ignore]

--- a/lib/galaxy/model/__init__.py
+++ b/lib/galaxy/model/__init__.py
@@ -10084,12 +10084,12 @@ class PSAAssociation(Base, AssociationMixin, RepresentById):
     __tablename__ = "psa_association"
 
     id: Mapped[int] = mapped_column(primary_key=True)
-    server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment]
-    handle: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment]
-    secret: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment]
-    issued: Mapped[Optional[int]]  # type:ignore[assignment]
-    lifetime: Mapped[Optional[int]]  # type:ignore[assignment]
-    assoc_type: Mapped[Optional[str]] = mapped_column(VARCHAR(64))  # type:ignore[assignment]
+    server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
+    handle: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
+    secret: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
+    issued: Mapped[Optional[int]]  # type:ignore[assignment, unused-ignore]
+    lifetime: Mapped[Optional[int]]  # type:ignore[assignment, unused-ignore]
+    assoc_type: Mapped[Optional[str]] = mapped_column(VARCHAR(64))  # type:ignore[assignment, unused-ignore]
 
     # This static property is set at: galaxy.authnz.psa_authnz.PSAAuthnz
     sa_session = None
@@ -10146,8 +10146,8 @@ class PSACode(Base, CodeMixin, RepresentById):
     __table_args__ = (UniqueConstraint("code", "email"),)
 
     id: Mapped[int] = mapped_column(primary_key=True)
-    email: Mapped[Optional[str]] = mapped_column(VARCHAR(200))  # type:ignore[assignment]
-    code: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment]
+    email: Mapped[Optional[str]] = mapped_column(VARCHAR(200))  # type:ignore[assignment, unused-ignore]
+    code: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment, unused-ignore]
 
     # This static property is set at: galaxy.authnz.psa_authnz.PSAAuthnz
     sa_session = None
@@ -10173,9 +10173,9 @@ class PSANonce(Base, NonceMixin, RepresentById):
     __tablename__ = "psa_nonce"
 
     id: Mapped[int] = mapped_column(primary_key=True)
-    server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment]
-    timestamp: Mapped[Optional[int]]  # type:ignore[assignment]
-    salt: Mapped[Optional[str]] = mapped_column(VARCHAR(40))  # type:ignore[assignment]
+    server_url: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
+    timestamp: Mapped[Optional[int]]  # type:ignore[assignment, unused-ignore]
+    salt: Mapped[Optional[str]] = mapped_column(VARCHAR(40))  # type:ignore[assignment, unused-ignore]
 
     # This static property is set at: galaxy.authnz.psa_authnz.PSAAuthnz
     sa_session = None
@@ -10209,10 +10209,10 @@ class PSAPartial(Base, PartialMixin, RepresentById):
     __tablename__ = "psa_partial"
 
     id: Mapped[int] = mapped_column(primary_key=True)
-    token: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment]
-    data: Mapped[Optional[str]] = mapped_column(TEXT)  # type:ignore[assignment]
-    next_step: Mapped[Optional[int]]  # type:ignore[assignment]
-    backend: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment]
+    token: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment, unused-ignore]
+    data: Mapped[Optional[str]] = mapped_column(TEXT)  # type:ignore[assignment, unused-ignore]
+    next_step: Mapped[Optional[int]]  # type:ignore[assignment, unused-ignore]
+    backend: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment, unused-ignore]
 
     # This static property is set at: galaxy.authnz.psa_authnz.PSAAuthnz
     sa_session = None
@@ -10252,12 +10252,14 @@ class UserAuthnzToken(Base, UserMixin, RepresentById):
 
     id: Mapped[int] = mapped_column(primary_key=True)
     user_id: Mapped[Optional[int]] = mapped_column(ForeignKey("galaxy_user.id"), index=True)
-    uid: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment]
-    provider: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment]
+    uid: Mapped[Optional[str]] = mapped_column(VARCHAR(255))  # type:ignore[assignment, unused-ignore]
+    provider: Mapped[Optional[str]] = mapped_column(VARCHAR(32))  # type:ignore[assignment, unused-ignore]
     extra_data: Mapped[Optional[bytes]] = mapped_column(MutableJSONType)
     lifetime: Mapped[Optional[int]]
     assoc_type: Mapped[Optional[str]] = mapped_column(VARCHAR(64))
-    user: Mapped[Optional["User"]] = relationship(back_populates="social_auth")  # type:ignore[assignment]
+    user: Mapped[Optional["User"]] = relationship(
+        back_populates="social_auth"
+    )  # type:ignore[assignment, unused-ignore]
 
     # This static property is set at: galaxy.authnz.psa_authnz.PSAAuthnz
     sa_session = None

